api/v1alpha4/common_types.go

```diff
@@ -29,6 +29,10 @@ const (
 	// ClusterTopologyLabelName is the label set on all the object which are managed as part of a ClusterTopology.
 	ClusterTopologyLabelName = "cluster.x-k8s.io/topology"
 
+	// MachineDeploymentTopologyLabelName is the label set on the generated  MachineDeployment objects
+	// to track the name of the machine deployment topology it represents.
+	MachineDeploymentTopologyLabelName = "cluster.x-k8s.io/topology/machine-deployment-name"
```

Comment:
Should it maybe rather be `ClusterTopologyMachineDeploymentLabelName`? (because of constant namespacing and because the label is part of the cluster topology feature)

Reply:
/cc @fabriziopandini @killianmuldoon

Reply:
+1 to the suggested name & to the label change

Reply:
Seems like the right thing to do to avoid confusion +1

Reply:
ACK'd